src/main/java/com/google/cloud/datafusion/plugin/ariba/source/AribaBatchSource.java

```diff
@@ -0,0 +1,124 @@
+import com.google.cloud.datafusion.plugin.ariba.source.util.Util;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.batch.BatchSourceContext;
+import org.apache.hadoop.io.LongWritable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+
+/**
+ * Batch Source that reads data from an Ariba system.
+ * <p>
+ * LongWritable is the first parameter because that is the key used by Hadoop's
+ * {@code TextInputFormat}. Similarly, Text is the second parameter because that
+ * is the value used by Hadoop's {@code TextInputFormat}.
+ * {@code StructuredRecord} is the third parameter because that is what the
+ * source will output. All the plugins included with Hydrator operate on
+ * StructuredRecord.
+ */
+
+@Plugin(type = BatchSource.PLUGIN_TYPE)
+@Name(AribaBatchSource.NAME)
+@Description("Reads data from Sap Ariba.")
+public class AribaBatchSource extends BatchSource<LongWritable, StructuredRecord, StructuredRecord> {
+
+  public static final String NAME = "Ariba";
+  private static final Logger LOGGER = LoggerFactory.getLogger(AribaBatchSource.class);
+  private final AribaPluginConfig pluginConfig;
+
+  public AribaBatchSource(AribaPluginConfig pluginConfig) {
+    this.pluginConfig = pluginConfig;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    LOGGER.trace("Initiating Ariba batch source configure pipeline");
+    FailureCollector failureCollector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
+    pluginConfig.validatePluginParameters(failureCollector);
+    AribaServices aribaServices = new AribaServices(pluginConfig);
+
+    if (pluginConfig.isSchemaBuildRequired()) {
+      try {
+        JsonNode jsonNode = aribaServices.getAccessToken();
+        // Get metadata if connection is successful
+        if (jsonNode != null) {
+          pipelineConfigurer.getStageConfigurer().setOutputSchema(
+            aribaServices.buildOutputSchema(jsonNode.get("access_token").asText()));
+        }
+      } catch (IOException exception) {
+        failureCollector.addFailure(exception.getMessage(), null);
+        failureCollector.getOrThrowException();
+      } catch (AribaException exception) {
+        attachFieldWithError(exception, failureCollector);
+        failureCollector.getOrThrowException();
+      }
+    } else {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(null);
+    }
+  }
+
+  @Override
+  public void prepareRun(BatchSourceContext context) throws Exception {
+
```

Comment:
this needs to be added

Reply:
This will be added in the next PR, just keep the volume of PR low we have segregated it into two parts.
It will be tracked under this ticket. https://cdap.atlassian.net/browse/PLUGIN-1172